server/oc_cmds.go

```diff
@@ -43,29 +47,64 @@ func ShowValidatorCmd() *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			serverCtx := GetServerContextFromCmd(cmd)
 			cfg := serverCtx.Config
-
-			privValidator := pvm.LoadFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile())
-			pk, err := privValidator.GetPubKey()
-			if err != nil {
-				return err
-			}
-			sdkPK, err := cryptocodec.FromOcPubKeyInterface(pk)
-			if err != nil {
-				return err
-			}
-			clientCtx := client.GetClientContextFromCmd(cmd)
-			bz, err := clientCtx.Codec.MarshalInterfaceJSON(sdkPK)
-			if err != nil {
-				return err
-			}
-			fmt.Println(string(bz))
-			return nil
+			return showValidator(cmd, cfg)
 		},
 	}
 
 	return &cmd
 }
 
+func showValidator(cmd *cobra.Command, config *cfg.Config) error {
+	var pv types.PrivValidator
+	if config.PrivValidatorListenAddr != "" {
+		chainID, err := LoadChainID(config)
+		if err != nil {
+			return err
+		}
+		serverCtx := GetServerContextFromCmd(cmd)
+		log := serverCtx.Logger
+		pv, err = node.CreateAndStartPrivValidatorSocketClient(config.PrivValidatorListenAddr, chainID, log)
+		if err != nil {
+			return err
+		}
+	} else {
+		keyFilePath := config.PrivValidatorKeyFile()
+		if !ostos.FileExists(keyFilePath) {
+			return fmt.Errorf("private validator file %s does not exist", keyFilePath)
+		}
+		pv = pvm.LoadFilePV(keyFilePath, config.PrivValidatorStateFile())
+	}
+
+	pubKey, err := pv.GetPubKey()
+	if err != nil {
+		return fmt.Errorf("can't get pubkey: %w", err)
+	}
+
+	bz, err := osjson.Marshal(pubKey)
+	if err != nil {
+		return fmt.Errorf("failed to marshal private validator pubkey: %w", err)
+	}
+
+	fmt.Println(string(bz))
+	return nil
+}
+
+func LoadChainID(config *cfg.Config) (string, error) {
```

Comment:
This function seems to be used only in this file. Why does it need to be public?
(You might call a private function in the same package, even for testing purposes.)

Reply:
I'm at a loss as well. Packages are separate for test files and others, and this test belongs to the `server_test` package. And this is tested by calling a private function in `util_test.go`. Therefore, this function could not be called from the test function unless it was made public.

Reply:
If the `server_test` package is used for that reason, I think it would be better to make the private functions in `util_test.go` public,  if the impact isn't too large.

Reply:
It's certainly a good idea to make the test function public. Thanks for your suggestion.